crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs

```diff
@@ -193,7 +195,8 @@ pub fn invalid_string_characters(
                 '\u{200b}' => ("\\u200b", InvalidCharacterZeroWidthSpace.into()),
                 _ => unreachable!(),
             };
-            let location = helpers::to_absolute(Location::new(row + 1, column), start);
+            let column_chars = line[..column_bytes].chars().count();
```

Comment:
@MichaReiser I've found this by guessing, i'm not sure if that's the correct way of computing offsets

Reply:
It's the correct way for computing char offsets, which is slow. That's why I want to change `Location` to use byte offsets directly so that it simply becomes `line.len()` which is `O(1)`

The current solution has a worst-case complexity of `O(n^2)`  because `line[..column_bytes].chars().count()` iterates from the beginning of the text for every match. I would recommend not using `match_indices` and iterate over `chars` directly. 

```rust
let mut char_offset = 0;
for (byte_offset, char) in line.char_indices() {
	let replacement = match char {
		...patterns
		_ => {
			char_offset += 1;
			continue;
		}
	};

	// .. rest

	char_offset += 1;
}

```

Reply:
updated, char() works